pkg/index/index.go

```diff
@@ -188,27 +190,95 @@ func (m *MergedIndexes) Merge(indexes ...*Index) {
 }
 
 // SearchByKeywords search for entries matching the given keywords in MergedIndexes.
-func (i *Index) SearchByKeywords(keywords ...string) []*Entry {
-	var result []*Entry
-	keywordSet := make(map[string]bool)
-
-	for _, keyword := range keywords {
-		keywordSet[keyword] = true
-	}
+// minScore is the minimum score to consider a match between a name of an artifact and a keyword.
+// if minScore is not reached, we fallback to a simple partial matching on keywords.
+func (i *Index) SearchByKeywords(minScore float64, keywords ...string) []*Entry {
+	var matches []*Entry
```

Comment:
Can't we use a  set of string here (ie: map[string]struct{}) and then return its keys, instead of deduplicating keys below?

Reply:
What do you think about it now? :)